lib/libc/newlib/include/stdint.h

```diff
@@ -13,12 +13,15 @@
  */
 
 #include <newlib.h>
+#include <stddef.h>
 
 #ifdef __NEWLIB__
 /* Has this header.  Older versions do it in <stdint.h>. */
 #include <sys/_stdint.h>
 #endif /* __NEWLIB__ */
 
+void qsort_r(void *base, size_t nmemb, size_t size,
+	     int (*compar)(const void *, const void *, void *), void *arg);
```

Comment:
As per https://github.com/zephyrproject-rtos/zephyr/pull/57619#issuecomment-1536683981, in order to support downstream applications that may choose to pull in a different flavour (e.g. BSD) of this function, it may be desirable to pull in the full definition here:

https://github.com/zephyrproject-rtos/newlib-cygwin/blob/4e150303bcc1e44f4d90f3489a4417433980d5ff/newlib/libc/include/stdlib.h#L308-L318

Reply:
I think we should cross that bridge if/when someone wants to do that.